src/macaron/build_spec_generator/reproducible_central/reproducible_central.py

```diff
@@ -0,0 +1,355 @@
+            except ValueError:
+                continue
+
+            # TODO: What happen if we report multiple build tool in the database.
+            return macaron_build_tool_name
+
+    return None
+
+
+def _gen_reproducible_central_build_spec(
+    build_info: RcInternalBuildInfo,
+    patches: Mapping[
+        PatchCommandBuildTool,
+        Mapping[str, PatchValueType | None],
+    ],
+) -> str | None:
+    """Return the RC's Buildspec content from a ``RcInternalBuildInfo`` instance.
+
+    This function will perform necessary validation on the data captured within ``build_info`` to make sure
+    it has enough information for generating the output BuildSpec.
+
+    This function will use the information available in ``build_info`` to populate the file.
+    For example, the GAV coordinate can be obtained from ``build_info.purl``.
+
+    The ``patches`` mapping will be used for patching the build command in the `command` section of the Buildspec
+    output.
+
+    The function will return the Buildspec file content as string or None if there is an error.
+    """
+    extra_comments = []
+
+    purl = build_info.purl
+    logger.debug(
+        "Generating build spec for %s with command patches:\n%s",
+        purl,
+        pformat(patches),
+    )
+
+    group = purl.namespace
+    artifact = purl.name
+    version = purl.version
+    if group is None or version is None:
+        logger.error("Missing group and/or version for purl %s.", purl.to_string())
+        return None
+
+    extra_comments.append(f"Input PURL - {purl}")
+
+    macaron_build_tool_name = _get_macaron_build_tool_name(build_info.build_tool_facts)
+    if not macaron_build_tool_name:
+        logger.error(
+            "The PackageURL %s doesn't have any build tool that we support for generating RC buildspec. It has %s.",
+            purl.to_string(),
+            [(fact.build_tool_name, fact.language) for fact in build_info.build_tool_facts],
+        )
+        return None
+
+    rc_build_tool_name = None
+    if macaron_build_tool_name == _MacaronBuildToolName.MAVEN:
+        rc_build_tool_name = _ReproducibleCentralBuildToolName.MAVEN
+    elif macaron_build_tool_name == _MacaronBuildToolName.GRADLE:
+        rc_build_tool_name = _ReproducibleCentralBuildToolName.GRADLE
+    if not rc_build_tool_name:
+        logger.critical("%s is not supported to generate RC's buildspec.", macaron_build_tool_name.value)
+        return None
+
+    # Set the default build command and jdk version.
+    # The default build command depends on the build tool, while the default jdk version
+    # is 8.
+    final_build_command_seq = _get_default_build_command_sequence(
+        macaron_build_tool_name=macaron_build_tool_name,
+        patches=patches,
+    )
+    if not final_build_command_seq:
+        logger.critical(
+            "Cannot generate a default build command for %s",
+            purl,
+        )
+        return None
+    final_jdk_version = "8"
```

Comment:
I think we should always try to obtain the jdk version from maven central jar, and only fall back to `8` if we cannot do so. Please feel free to let me know if this is desireable.

Reply:
From this commit - https://github.com/oracle/macaron/pull/1115/commits/9e60c860c86208e77e8ff1644c10c1659044f24b we will always look for JDK version from Maven Central Jar.